eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+import java.util.regex.Pattern;
+
+/**
+ * Requirement of Javadoc annotation @codesnippet check:
+ *  (1) Use '{@codesnippet ...}' instead of '<code>', '<pre>'.
+ *  (2) No multiple lines span at the '{@code ...}'.
+ *  (3) naming pattern of codesnippet":
+ *    1, For package name, class name, they should all be lower case.
+ *       For method name,  the first letter of method name should be lower case.
+ *       For parameters, they should only be letters. Concatenate all parameters with dash character.
+ *    2, Jonathon will add more requirement specs later.
+ */
+public class JavadocCodeSnippetCheck extends AbstractJavadocCheck {
+
+    private static final String CLASS_PATH_REGEX = "[a-z.]+";
+    private static final String METHOD_NAME_REGEX = "^[a-z][a-zA-Z]+";
+    private static final String PARAMETERS_REGEX = "[a-zA-Z-]+";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return getRequiredJavadocTokens();
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.HTML_ELEMENT_START,
+            JavadocTokenTypes.JAVADOC_INLINE_TAG
+        };
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode token) {
+        switch (token.getType()) {
+            case JavadocTokenTypes.HTML_ELEMENT_START:
+                checkHtmlElementStart(token);
+                break;
+            case JavadocTokenTypes.JAVADOC_INLINE_TAG:
+                checkJavadocInlineTag(token);
+                checkCodeSnippetNaming(token);
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Requirement of rules:
+     * (1) No usage of '<code>', '<pre>',  use '{@codesnippet ...} instead'.
+     * (2) No multiple lines span at the '{@code ...}'
+     *
+     * @param htmlElementStartNode
+     */
+    private void checkHtmlElementStart(DetailNode htmlElementStartNode) {
+        final DetailNode tagNameNode = JavadocUtil.findFirstToken(htmlElementStartNode, JavadocTokenTypes.HTML_TAG_NAME);
+        final String tagName = tagNameNode.getText();
+        if ("code".equals(tagName) || "pre".equals(tagName)) {
```

Comment:
This could be easier to check if you used a set. This would make it easier to extend if there are other HTML tags we forgot about. In addition, HTML tags are case insensitive, you're missing out on `<PRE>` cases, etc.
```java
class Check {
  private static final Set<String> INVALID_TAGS = Collections.unmodifiableSet("pre","code");
  void checkHtmlElement() {
      if (INVALID_TAGS.contains(tagName)) {
  // log
      }
  }
}
```